Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -136,3 +179,22 @@ void UPSSpotComponent::RefreshAmountOfUnlockedSkins(bool bApplySkin)
 		}
 	}
 }
+
+// Returns true if this is a current spot
+bool UPSSpotComponent::IsCurrentSpot()
+{
+	const APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
+
+	if (!ensureMsgf(PlayerCharacter, TEXT("ASSERT: [%i] %hs:\n'PlayerCharacter' is not valid!"), __LINE__, __FUNCTION__))
```

Comment:
- it's not recommended to add any `ensures` in Getter\Is functions, as they can be called in unexpected context or too early, that is okay, so such ensures will be triggered a lot.
- finally, function can be simplified by removing `If`-condition: just return result directly
```
bool UPSSpotComponent::IsCurrentSpot() const
{
	const APlayerCharacter* PlayerCharacter = UMyBlueprintFunctionLibrary::GetLocalPlayerCharacter();
	const FPlayerTag& PlayerTag = PlayerCharacter ? PlayerCharacter->GetPlayerTag() : FPlayerTag::None;
	return PlayerCharacter && GetMeshChecked().GetPlayerTag() == PlayerTag;
}
```

Reply:
done https://github.com/moinm3uw/ProgressionSystem/pull/63/commits/a44979756036940641fa7ea10e73ab9cd359afc9